packages/type-primitives/src/AccountIndex.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017-2018 @polkadot/primitives authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { AccountIndex, AccountIndexDecoder } from './types';
+
+import BN from 'bn.js';
+import bnToBn from '@polkadot/util/bn/toBn';
+
+class AccountIndexImpl implements AccountIndex {
```

Comment:
This needs to probably extends a "Number" class that has all the basics - only things different between AccountIndex, BlockNumber, Balance, u32, u64, etc. is the actual sizes.

so ...

```
class Number {
  constructor (bitLength: number = 32 | 64 | 128 | 256) {
    this._bitLength = bitLength;
  }
...
}

class AccountIndex extends Number {
  constructor () {
    super(64);
  }
...
}
```